zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java

```diff
@@ -376,25 +375,33 @@ public boolean runAllParagraphs(String noteId,
       return false;
     }
 
-    if (paragraphs == null) {
-      paragraphs = gson.fromJson(gson.toJson(note.getParagraphs()), new TypeToken<List>(){}.getType());
-    }
-
     note.setRunning(true);
     try {
-      for (Map<String, Object> raw : paragraphs) {
-        String paragraphId = (String) raw.get("id");
-        if (paragraphId == null) {
-          continue;
+      if (paragraphs != null) {
+        // run note via the data passed from frontend
+        for (Map<String, Object> raw : paragraphs) {
+          String paragraphId = (String) raw.get("id");
+          if (paragraphId == null) {
+            continue;
+          }
+          String text = (String) raw.get("paragraph");
+          String title = (String) raw.get("title");
+          Map<String, Object> params = (Map<String, Object>) raw.get("params");
+          Map<String, Object> config = (Map<String, Object>) raw.get("config");
+
+          if (!runParagraph(noteId, paragraphId, title, text, params, config, false, true,
+                  context, callback)) {
+            // stop execution when one paragraph fails.
```

Comment:
I maybe would add more logging around `noteId` and `paragraphId` into the `runParagraph`. WDYT?